lib/modules/measure/MeasureService.ts

```diff
@@ -300,10 +285,18 @@ export class MeasureService {
     }
   }
 
+  // @todo there shouldn't be any logic related to asset historization here, but no other choices for now. It needs to be re-architected
+  /**
+   * Update asset with each non-null non-computed measures.
+   *
+   * @returns An object of each asset state by timestamp.
+   */
   private updateAssetMeasures(
     asset: KDocument<AssetContent>,
     measurements: MeasureContent[]
-  ) {
+  ): Map<number, KDocument<AssetContent<any, any>>> {
```

Comment:
cannot create a dedicated object instead of this ugly map ?

Reply:
I added [a comment](https://github.com/kuzzleio/kuzzle-device-manager/pull/290/commits/545c79a5bc3cdd0abb2613a965fccdd4e195fb51#diff-064fc947b9c2a22f50298738d3bf7f5cec0172e69d7c6abbd64b5c07409c5c1fR298) to indicate why we use a Map (preserve insertion order).

We can use a type alias but it wouldn't be used only in one file AND it wouldn't help clarity
I don't see better ways to do it:
```ts
type AssetStates = Map<number, KDocument<AssetContent<any,any>>
```
Moreover, we would have things like:
```ts
const assetStates: AssetStates = new Map<number, KDocument<AssetContent>>();
```

I'm confused, perhaps we should just leave it like this...

Ping @Aschen

Reply:
Why not using an array of tuples? `Array<[measuredAt, asset]>`